### PR TITLE
fix: check for display manager before adding password manager hint

### PR DESCRIPTION
**Pull request by hxu1**
As per #1308 and #1307, copying composite objects are not working as expected on Wayland. 

I checked around the Internet, and it looks like some other projects are also having difficulties with composite objects and Wayland (example: https://github.com/flameshot-org/flameshot/issues/2848)

Adding a check for the display manager and only add the password manager hint when the display manager is X11.

### CODE REVIEW

**Comment by hxu1** [comment]
Hi @ronys, thanks for the review. I have addressed the review comments, can you please check?

File: src/ui/wxWidgets/Clipboard.cpp [hunk 1]
```diff
@@ -107,10 +107,14 @@ bool Clipboard::SetData(const StringX &data)
   bool res = false;
   if (wxTheClipboard->Open()) {
 #if defined(__UNIX__) && !defined(__WXOSX__)
```
[the remark this answers is not in the document]
Comment (hxu1) [reply src/ui/wxWidgets/Clipboard.cpp:109]: Added (https://github.com/pwsafe/pwsafe/pull/1310/files#diff-de2d28dd8c542b142792281c1c4219b6a1c03d6869ca8cb1d73974f2ec1ca3ecR110)

File: src/ui/wxWidgets/wxUtilities.h [hunk 2]
```diff
@@ -427,8 +427,14 @@ namespace wxUtilities
...
    * @brief Checks environment variable 'XDG_SESSION_TYPE' for display manager 'x11'.
+   *
+   * @return whether the environment variable is set to 'x11'.
+   */
+  bool IsDisplayManagerX11();
+  /**
+   * @brief Checks if virtual keyboard is supported.
    * 
-   * @return true if environment variable is set to 'x11' on Linux,
+   * @return true if IsDisplayManagerX11() returns true on Linux,
```
[the remark this answers is not in the document]
Comment (hxu1) [reply src/ui/wxWidgets/wxUtilities.h]: Updated.

**Comment by igor2x** [comment]
@ronys, after merging this PR I suggest to release v1.19.1, because without this fix current v1.19 is pretty useless on Wayland. Double click or right click and Copy Password does not copy password on Fedora like reported in [issue 1307](https://github.com/pwsafe/pwsafe/issues/1307) and crashes Ubuntu like reported in [issue 1308](https://github.com/pwsafe/pwsafe/issues/1308). Currently users using v1.19 on Linux are required to see the password in Password Manager and then manually type in any password form.

After this PR is merged I also suggest to create flatpak like described in [issue 1320.](https://github.com/pwsafe/pwsafe/issues/1320)

**Comment by ronys** [comment]
I've tested this on Ubuntu 24 and Fedora 40, in both nothing is copied to the clipboard.
Under Ubuntu the display manager no longer crashes, though.

**Comment by igor2x** [comment]
@ronys, I have installed [@hxu1 commit](https://github.com/pwsafe/pwsafe/actions/runs/9706435059/artifacts/1647074961) Artifact from Github on Ubuntu 2**2**.04 on Wayland and I see problem is fixed: password is copied and I can paste password to text editor.

@ronys, what do you use Wayland or Xorg? I can't test on Xorg, because my laptop does not work under Xorg (years old problem on my laptop).

**Comment by igor2x** [comment]
I did another test on Ubuntu 2**4**.04 on Wayland (the same laptop but under VirtualBox):
1. Downloaded https://github.com/hxu1/pwsafe/archive/refs/heads/FixCopyPasteWayland.zip
2. Then followed [Linux build instructions](https://github.com/pwsafe/pwsafe/blob/master/README.LINUX.DEVELOPERS.md) and build program and installed deb file.
3. Tested double click on password and paste works fine in text editor.

It works fine for me... on Wayland.

**Comment by nobugshere** [comment]
Works okay for me too.  Tested on the following:
Fedora 40 ARM64, Xfce/X11, gcc 14.1.1, wx 3.2.4, GTK 3.24.42
Fedora 40 ARM64, KDE/Wayland, gcc 14.1.1, wx 3.2.4, GTK 3.24.42
Ubuntu 24.04 ARM64, Gnome/Wayland, gcc 13.2.0, wx 3.2.4, GTK 3.24.41

**Comment by hxu1** [comment]
Hi @ronys, I've tested this on Kubuntu 24.04, copy and paste are working as expected for both X11 and Wayland. Can you please share your setup (display manager and desktop environment)?

**Comment by ronys** [comment]
Thanks @igor2x , @nobugshere and @hxu1 . Looks like a reboot was needed between reproducing the bug and testing the fix.
Merging and releasing this as 1.19.1.

**Comment by igor2x** [comment]
Ronys,
thanks a lot for merge PR.

I think we should release new flatpak with Password Safe v1.19.1 I have
tested it and wrote instructions in
https://github.com/pwsafe/pwsafe/issues/1320.

Regards

**Pull request merged**